components/controls/ClipboardIconButton.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Sonar UI Common
+ * Copyright (C) 2019-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import * as classNames from 'classnames';
+import Tooltip from './Tooltip';
+import { Button } from './buttons';
+import ClipboardBase from './ClipboardBase';
+import { translate } from '../../helpers/l10n';
+
+interface State {
+  isCopied: boolean;
+}
+
+export default class ClipboardIconButton extends ClipboardBase {
+  copyButton?: HTMLElement | null;
+  mounted = false;
+  state: State = { isCopied: false };
+
+  showTooltip = () => {
```

Comment:
And if you use the `ButtonIcon` this definitely won't handle showing the tooltip or not. So let's make it more generic and rename it here and in the base to something like `handleSuccessCopy`.
In this component all we will do is change the text and revert it 1sec later, and leave all the display aspect of the tooltip to `ButtonIcon`.